up-l1/someip.adoc

```diff
@@ -0,0 +1,265 @@
+
+==== UUIDs
+
+link:../basics/uuid.adoc[uProtocol UUID] specifications create a unique identifier for each message along with timestamp information.
+
+The ID is used for correlate between request and response as well. *SOME/IP* instead defines the `Request ID` as 16 bit `Client ID` + 16 bit `Session ID` (that is incremented). `[PRS_SOMEIP_00046]`
+
+When messages are converted *SOME/IP* to/from *uProtocol*, care must be taken to ensure that the *SOME/IP* ID and uProtocol UUID are properly mapped, especially when corelating a request to a response.
+
+* Generated *SOME/IP* Events *MUST* set the 16 bit `Client ID` to 0 per `[PRS_SOMEIP_00925]`
+* Generated *SOME/IP* Responses *MUST* auto-populate the `Request ID` cached from the request message, into the response message and then flush the entry in the cache.
+
+Further details of the usage of IDs for the various message types can be found in the next section.
+
+==== Message Type
+
+<<umessagetype-mapping>> table below maps of uProtocol messages to `[PRS_SOMEIP_00055]` *SOME/IP* message types.
+
+.UMessageType Mapping
+[umessagetype-mapping, cols="1,1,2"]
+|===
+| UMessageType | SOME/IP Type | Details
+
+a| `UMESSAGE_TYPE_PUBLISH` a| `NOTIFICATION` | Publish
+
+a| `UMESSAGE_TYPE_REQUEST` a| `REQUEST` | Requests
+
+a| `UMESSAGE_TYPE_RESPONSE` a| `RESPONSE` or `ERROR` | Responses or error has occurred while attempting to deliver the message or a service
+has thrown an exception
+
+a| `UMESSAGE_TYPE_NOTIFICATION` a| `NOTIFICATION` | Notification
+
+|===
+
+
+When receiving *uProtocol* initiated requests:
+
+* *MUST* cache the request `UAttributes` for a maximum of `ttl` so that it can be used to build a response `UAttributes` when receiving a response from *SOME/IP*
+
+```
+response.priority = request.priority
+response.reqid = request.id
+```
+
+When sending auto-generated *SOME/IP* REQUEST messages:
+
+  * *MUST* cache the message's `Request ID` to correlate with the RESPONSE message.
+  * Underlying *SOME/IP* library *SHOULD* handle `Request ID` updating.
+
+When receiving a *SOME/IP* initiated requests:
+
+  * *MUST* cache the *SOME/IP* `Request ID` as well as the generated `UAttributes` for the request messages so that the response can be translated back to a *SOME/IP* response message
+
+
+==== Communication Status
+
+<<commstatus-mapping>> below provides the mapping of link:../basics/uattributes.adoc[UAttributes] `commstatus` `UCode` codes to `[PRS_SOMEIP_0019]` *SOME/IP* error codes.
+
+.Error Code Mapping
+[commstatus-mapping, width=70%]
+|===
+| UCode | SOME/IP Error Codes
+
+a| `OK` a| `E_OK`
+a| `INVALID_ARGUMENT` a| `E_WRONG_MESSAGE_TYPE` or `E_UNKNOWN_METHOD`
+a| `DEADLINE_EXCEEDED` a| `E_TIMEOUT`
+a| `NOT_FOUND` a| `E_UNKNOWN_SERVICE`
+a| `UNAVAILABLE` a| `E_NOT_READY`
+a| `DATA_LOSS` a| `E_MALFORMED_MESSAGE`
+a| `INTERNAL` a| `E_NOT_REACHABLE`
+a| `UNKNOWN` a| `E_NOT_OK`
+a| `FAILED_PRECONDITION` a| `E_WRONG_PROTOCOL_VERSION` or `E_WRONG_INTERFACE_VERSION`
+|===
+
+
+=== UPayload
```

Comment:
@int0x27 UMessage no longer contains UPayload and there is no more reference vs vale. Now the UMessage contains the data in payload and the UPayloadFormat is stored in UAttributes (change made last week).

Reply:
I noticed it was referring to a missing entry and tried to figure it out. Maybe it will be better if you rewrite this section.
e.g. why using unspecified payload format (actually that was removed as well), instead of being `UPAYLOAD_FORMAT_SOMEIP`. 
Is there any difference between `UPAYLOAD_FORMAT_RAW` and `UPAYLOAD_FORMAT_SOMEIP`?